test/ng/compileSpec.js

```diff
@@ -7815,4 +7817,125 @@ describe('$compile', function() {
         expect(element.hasClass('fire')).toBe(true);
       }));
   });
+
+  describe('element replacement', function() {
+    it('should broadcast $destroy only on removed elements, not replaced', function() {
```

Comment:
Did you try to remove the `jqLite(firstElementToRemove).off('$destroy');` line and run unit tests? From what I remember the whole `skipDestroyOnNextJQueryCleanData` was added because the scope is reused on the replaced elements so removing this logic made existing unit tests fail.